src/frontend/src/optimizer/rule/index_selection.rs

```diff
@@ -205,11 +198,12 @@ impl IndexSelectionRule {
             .chain(new_predicate.into_iter())
             .collect_vec();
         let on = Condition { conjunctions };
-        let join = LogicalJoin::new(
+        let join = LogicalJoin::new_with_hint(
             index_scan.into(),
             primary_table_scan.into(),
             JoinType::Inner,
             on,
+            true,
```

Comment:
Looks a litte wierd to me. Is it reasonable to do index selection in logical optimization stage? 🤔

Reply:
Yes, It is hard when I try to add this flag. Let me think about another more elegant way. Currently, after index selection, our plan immediately enter `to_batch()` where maybe a better place to perform index selection.

Reply:
Finally I decide to move index selection to `to_batch()` of the `LogicalScan` where we can definitely convert logical join to an index lookup join without any flag. @fuyufjh please take look again.